src/engine/DepthOrderedCollection.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+#include <iterator>
+#include <map>
+#include <memory>
+#include <set>
+#include <vector>
+
+#include "IOrderedObject.hpp"
+
+namespace admirals {
+
```

Comment:
Needs a explanation of how the collection works as well as how stuff are sorted.